src/NavigationContainer.tsx

```diff
@@ -71,6 +78,32 @@ export default function NavigationContainer({
   const isTransactionActiveRef = React.useRef<boolean>(false);
   const isFirstMountRef = React.useRef<boolean>(true);
 
+  React.useImperativeHandle(ref, () => ({
+    dispatch: action => {
+      context.performTransaction(() => {
+        for (let i = 0; i < actionListeners.length; i++) {
+          actionListeners[i]({
+            type: '__DISPATCH_INTERNAL',
```

Comment:
Instead of a special type, what do you think about changing the signature of `onAction` function? Something like:

```js
function onAction(action: NavigationAction, source?: { root?: boolean, target?: string, navigator?: string });
```

I think it's better because otherwise we're leaking internals as anyone can call `dispatch` with such an action.